docs/contracts.puml

```diff
@@ -0,0 +1,64 @@
+@startuml
+
+class "TokenFactory" as factory {
+    totalTokenCount()
+    Get created token count
+    --
+    tokenOf(id)
+    Get token address by index
+    --
+    tokenAmountOf(address)
+    get token amount of creator
+    --
+    creatorTokenOf(creator, id)
+    get a creator token of specific id
+    --
+    createToken(creator, name, symbol, totalSupply, decimals, ratio, isMintable, lockupPeriod, enableStakeToToken)
+    Create new creator token, and then transfer tokens to owner and vesting contract
+}
+
+class "ABCToken" as abc {
+    inherit:
+    - ERC20
+    --
+    Sample creator token created by FanStockManager.
+    Multiple creator tokens are created actually.
+    --
+    name
+    Token name
+    --
+    symbol
+    Token symbol
+    --
+    int8 decimals
+    --
+    totalSupply()
+}
+
+class "TokenPool" as pool {
+    earned(account, token)
+    Get available new token amount to claim
+    --
+    stake(amount, token)
+    --
+    withdraw(amount, token)
+    --
+    claim(token)
+}
+
+class "Vesting" as vest {
```

Comment:
Is there any way to know how long the Lockup will last?

Reply:
I assume using public variables like uniswap `vestingEnd`.
https://github.com/Uniswap/governance/blob/eabd8c71ad01f61fb54ed6945162021ee419998e/contracts/TreasuryVester.sol#L14